pulsar-client/src/main/java/org/apache/pulsar/client/impl/transaction/TransactionCoordinatorClientImpl.java

```diff
@@ -73,8 +72,8 @@ public void start() throws TransactionCoordinatorClientException {
 
     @Override
     public CompletableFuture<Void> startAsync() {
-        if (STATE_UPDATER.compareAndSet(this, State.NONE, State.STARTING)) {
-            return pulsarClient.getLookup().getPartitionedTopicMetadata(TopicName.TRANSACTION_COORDINATOR_ASSIGN)
+        STATE_UPDATER.compareAndSet(this, State.NONE, State.STARTING);
```

Comment:
If we call the start method multiple times, we will get unexpected behavior here

Reply:
`TransactionCoordinatorClientImpl` is not a class exposed to the user. Its start method will only be called once in the PulsarClient's constructor.

Reply:
We can keep the current logic but simply throw another exception, like a IllegalStateException.

This way we can get rid of the redundant exception